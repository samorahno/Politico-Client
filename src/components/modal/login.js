
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { loginUser } from '../../actions';
import '../../styles/landing.css';

class Login extends Component {
    constructor (props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            errors: {}
        };
        this.modal = React.createRef();
        this.onChange = this.onChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }

    closeModal (e) {
        document.getElementById('myModal').style.display = 'none';
    }


    onChange (e) {
        this.setState({ [e.target.name]: e.target.value });
    }

    async onSubmit (e) {
        e.preventDefault();
        const { login } = this.props;
        const user = {
            email: this.state.email,
            password: this.state.password
        };
        const res = await login(user);
        console.log(res);
    }

    render () {
        return (
            <div>
                <div ref={this.modal} id="myModal" className="modal" style={{ display: this.props.showModal ? 'block' : 'none' }}>

                    <div className="modal-content">
                        <span className="close" onClick={this.closeModal}>&times;</span>
                        <form className="form-modal dd validate" onSubmit={this.onSubmit}>
                            <h1>Login</h1>
                            <div className = "success-msg" id = "success-msg-login" />
                            <div className = "warning-msg" id = "warning-msg-login" />
                            <div className = "error-msg" id = "error-msg-login" />
                            <div className = "info-msg" id = "info-msg-login" />
                            <div className="input-group">
                                <label>Email</label>
                                <input
                                    type="email"
                                    name="email"
                                    placeholder="Email"
                                    id="loginemail"
                                    value = {this.state.email}
                                    onChange = {this.onChange}
                                    required />
                            </div>
                            <div className="input-group">
                                <label>Password</label>
                                <input
                                    type = "password"
                                    name="password"
                                    placeholder="Password"
                                    id="loginpassword"
                                    minLength="5"
                                    maxLength="50"
                                    value = {this.state.password}
                                    onChange = {this.onChange}
                                    required
                                />
                            </div>

                            <div className="input-group">
                                <button type="submit" name="submit" className="btn" id="loginbtn" >Login</button>
                                <a href="sample.html" id="myBtn3">
                                    <p className="fp">forgot password?</p>
                                </a>
                            </div>

                        </form>
                    </div>

                </div>
            </div>
        );
    }
}

const mapDispatchToProps = {
    login: loginUser
};
export default connect(null, mapDispatchToProps)(Login);
