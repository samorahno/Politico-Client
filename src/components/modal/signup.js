
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import { withRouter } from 'react-router-dom';
import ToastMessage from '../common/ToastMessage';
import { signupUser } from '../../actions';
import '../../styles/landing.scss';

// const baseUrlSignup = 'https://samson-politico.herokuapp.com/api/v1/';
export class Signup extends Component {
    constructor (props) {
        super(props);
        this.state = {
            firstName: '',
            lastName: '',
            email: '',
            phoneNumber: '',
            password: '',
            password_confirm: '',
            error: ''
        };
        this.onChange = this.onChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
        this.closeModal = this.closeModal.bind(this);
    }
    closeModal (e) {
        document.getElementById('myModal2').style.display = 'none';
    }

    onChange (e) {
        this.setState({ [e.target.name]: e.target.value });
    }

    redirectUser () {
        const { history } = this.props;
        history.push('/dashboard');
    }


    async onSubmit (e) {
        e.preventDefault();
        var newUser = {
            firstname: this.state.firstName,
            lastname: this.state.lastName,
            email: this.state.email,
            phone: this.state.phoneNumber,
            password: this.state.password,
            confirmPassword: this.state.password_confirm

        };
        const { signup } = this.props;
        const isAuthenticated = await signup(newUser);
        if (!isAuthenticated.payload.status) {
            toast(<ToastMessage message="Registration Successful. Redirecting....." />, {
                type: 'success',
                closeButton: false,
                onClose: () => this.redirectUser(),
                hideProgressBar: true,
                autoClose: 0
            });
        } else {
            toast(<ToastMessage message={isAuthenticated.payload.error || isAuthenticated.payload.message} />, {
                type: 'error',
                closeButton: false,
                hideProgressBar: true,
                autoClose: 5000
            });
        }
    }
    render () {
        const { error } = this.state.error;
        return (
            <div>
                <div id="myModal2" className="modall" style={{ display: this.props.showModal ? 'block' : 'none' }}>

                    <div className="modal-contentl" id="modalcontent">
                        <span className="closel" onClick = {this.closeModal} >&times;</span>

                        <form className="form-modall ddl" onSubmit={this.onSubmit}>
                            <h1>Signup</h1>
                            <div className = "success-msg" id = "success-msg" />
                            <div className = "warning-msg" id = "warning-msg" />
                            <div className = "error-msg" id = "error-msg" />
                            <div className = "info-msg" id = "info-msg" />
                            {error && (<div className="error-msgl">{error}</div>)}
                            <div className="input-groupl">
                                <label>First Name</label>
                                <input
                                    type="text"
                                    name="firstName"
                                    placeholder="First Name"
                                    id="firstname"
                                    value={this.state.firstName}
                                    onChange={this.onChange}
                                />

                            </div>
                            <div className="input-groupl">
                                <label>Last Name</label>
                                <input
                                    type="text"
                                    name="lastName"
                                    placeholder="Last Name"
                                    id="lastname"
                                    value={this.state.lastName}
                                    onChange={this.onChange}
                                />

                            </div>
                            <div className="input-groupl">
                                <label>E-mail</label>
                                <input
                                    type="email"
                                    name="email"
                                    placeholder="E-mail"
                                    id="signupemail"
                                    value={this.state.email}
                                    onChange={this.onChange}
                                />
                            </div>
                            <div className="input-groupl">
                                <label>Phone Number</label>
                                <input
                                    type="number"
                                    name="phoneNumber"
                                    placeholder="Phone Number"
                                    id="phonenumber"
                                    value={this.state.phoneNumber}
                                    onChange={this.onChange}
                                />
                            </div>
                            <label style={{ width: '33%' }}>Password / </label><label style={{ width: '33%' }}>Confirm Password</label>
                            <div className="input-groupl">
                                <input
                                    type="password"
                                    name="password"
                                    placeholder="Password"
                                    id="signuppassword"
                                    style={{ width: '33%' }}
                                    value={this.state.password}
                                    onChange={this.onChange}
                                />
                                <input
                                    type="password"
                                    name="password_confirm"
                                    placeholder="Confirm Password"
                                    id="confirmsignuppassword"
                                    style={{ width: '33%' }}
                                    value={this.state.password_confirm}
                                    onChange={this.onChange}
                                />
                            </div>
                            <div className="input-groupl">
                                <button type="submit" name="submit" className="btnl" id="signupbtn">Register</button>
                            </div>

                        </form>
                    </div>

                </div>
            </div>
        );
    }
}

const mapDispatchToProps = {
    signup: signupUser
};

export const mapStateToProps = (state) => ({
    userData: state.auth
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Signup));
