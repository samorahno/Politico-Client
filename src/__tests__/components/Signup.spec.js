import React from 'react';
import { shallow } from 'enzyme';
import { Signup } from '../../components/modal/signup';

const props = {
    history: {
        push: jest.fn()
    },
    signup: jest.fn(),
    isAuthenticated: {
        payload: {
            status: true
        }
    }
};

let signupData = {
    firstname: 'sola',
    lastname: 'bimbp',
    email: 'b@gmail.com',
    phone: '0803343333',
    password: 'jhjhjkhkj',
    confirmPassword: 'jhjhjkhkj'
};

describe('Signup Component', () => {
    it('should render Signup component correctly', () => {
        const component = shallow(<Signup {...props}/>);
        expect(component).toMatchSnapshot();
    });

    it('should return onchange for firstname', () => {
        const component = shallow(<Signup {...props}/>);
        component.find('input').at(0).simulate('change', {
            target: { value: signupData.firstname }
        });

        component.find('input').at(0).simulate('focus');
        expect(component).toMatchSnapshot();
    });

    it('should return onchange for lastname', () => {
        const component = shallow(<Signup {...props}/>);
        component.find('input').at(1).simulate('change', {
            target: { value: signupData.lastname }
        });

        component.find('input').at(1).simulate('focus');
        expect(component).toMatchSnapshot();
    });

    it('', () => {
        const component = shallow(<Signup {...props}/>);
        component.find('form').simulate('submit', {
            preventDefault: () => {}
        });
        component.setState({
            firstname: signupData.firstName,
            lastname: signupData.lastName,
            email: signupData.email,
            phone: signupData.phoneNumber,
            password: signupData.password,
            confirmPassword: signupData.password_confirm
        });
        expect(component).toMatchSnapshot();
    });
});
