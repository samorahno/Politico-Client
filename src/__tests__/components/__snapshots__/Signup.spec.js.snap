// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component  1`] = `ShallowWrapper {}`;

exports[`Signup Component should render Signup component correctly 1`] = `ShallowWrapper {}`;

exports[`Signup Component should return onchange for firstname 1`] = `ShallowWrapper {}`;

exports[`Signup Component should return onchange for lastname 1`] = `ShallowWrapper {}`;
